src/main/java/teammates/storage/sqlapi/FeedbackResponseCommentsDb.java

```diff
@@ -110,6 +114,56 @@ public FeedbackResponseComment getFeedbackResponseCommentForResponseFromParticip
         return HibernateUtil.createQuery(cq).getResultStream().findFirst().orElse(null);
     }
 
+    /**
+     * Updates a feedback response comment by {@link FeedbackResponseComment}.
+     *
+     * @return updated feedback response comment
+     * @throws InvalidParametersException if attributes to update are not valid
+     * @throws EntityDoesNotExistException if the comment cannot be found
+     */
+    public FeedbackResponseComment updateFeedbackResponseComment(FeedbackResponseComment newFeedbackResponseComment)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        assert newFeedbackResponseComment != null;
+
+        FeedbackResponseComment oldFeedbackResponseComment = getFeedbackResponseComment(newFeedbackResponseComment.getId());
+        if (oldFeedbackResponseComment == null) {
+            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT + newFeedbackResponseComment);
+        }
+
+        newFeedbackResponseComment.sanitizeForSaving();
+        if (!newFeedbackResponseComment.isValid()) {
+            throw new InvalidParametersException(newFeedbackResponseComment.getInvalidityInfo());
+        }
+
+        // update only if change
+        boolean hasSameAttributes =
+                this.<Long>hasSameValue(newFeedbackResponseComment.getId(), oldFeedbackResponseComment.getId())
+                && this.<String>hasSameValue(
+                    newFeedbackResponseComment.getCommentText(), oldFeedbackResponseComment.getCommentText())
+                && this.<List<FeedbackParticipantType>>hasSameValue(
+                    newFeedbackResponseComment.getShowCommentTo(), oldFeedbackResponseComment.getShowCommentTo())
+                && this.<List<FeedbackParticipantType>>hasSameValue(
+                    newFeedbackResponseComment.getShowGiverNameTo(), oldFeedbackResponseComment.getShowGiverNameTo())
```

Comment:
Can I check why other attributes like `giver`, `isVisibilityFollowingFeedbackQuestion` and `isCommentFromFeedbackParticipant` don't have to be checked?

Reply:
Good point, I copied them over and missed these attributes, will add them in

Reply:
Shifted to #12475